effects/taxes_effect.py

```diff
@@ -0,0 +1,105 @@
+from typing import Tuple
+
+import message_text_config as msg
+import utils
+from citizens.citizen import Citizen
+from citizens.player import Player
+from citizens.spy import Spy
+from game import Game
+
+from effects.effect import Effect
+
+
+class TaxesEffect(Effect):
+    def __init__(self, context: 'Context', name: str,
+                 creator: Citizen) -> None:
+        super().__init__(context, name, creator, 14)
+
+    def _activate_impl(self) -> bool:
+        target_number = utils.read_target_number(
+            self.context, msg.TaxesMessages.ACTIVATION_CHOOSE_TARGET,
+            self._validate)
+        self.targets.append(self.game.citizens[target_number - 1])
+
+        self.user_interaction.save_active(
+            msg.TaxesMessages.ACTIVATION_SUCCESS.format(self.targets[0].name))
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        if isinstance(self.targets[0], Player):
+            if self.activation_round == self.game.round_number:
+                return False
+            else:
+                while (True):
+                    if self._choose_payment():
```

Comment:
You can move infinit cycle to _choose_payment method

Reply:
Done